src/data_readers/data_reader_merge_features.cpp

```diff
@@ -43,7 +43,7 @@ data_reader_merge_features::data_reader_merge_features(
   const data_reader_merge_features& other) :
   generic_compound_data_reader(other),
   m_data_size(other.m_data_size) {
-  if(other.m_label_reader != nullptr) 
+  if(other.m_label_reader != nullptr)
```

Comment:
pointers implicitly cast to `bool`? Why `!= nullptr`?

Reply:
We use the `if (ptr != nullptr)` idiom throughout the codebase. I think it's marginally clearer, but perhaps it is excessive.

Reply:
Eh, it's not a big deal. It just stood out as overly verbose. For folks not using IDEs, I suppose it suggests the pointer nature of `m_label_reader`, which is good documentation. I rescind my complaint.